Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -381,7 +381,7 @@ extension ChatChannel {
         let reads: [ChatChannelRead] = try dto.reads.map { try $0.asModel() }
         
         let unreadCount: () -> ChannelUnreadCount = {
-            guard let currentUser = context.currentUser else { return .noUnread }
+            guard dto.isValid, let currentUser = context.currentUser else { return .noUnread }
```

Comment:
And what about checking if it is deleted as well? If the Channel is deleted, we are not interested in fetching its data right?

Reply:
`isValid` will return false if the dto is deleted.